### ISSUE TITLE
Ensure buttons have discernible text

### ISSUE THREAD

**Issue by mgifford** [mastodon/mastodon#20443]
### Steps to reproduce the problem

On the notification page in this instance https://mastodon.social/notifications the button does not have text. 

### Expected behaviour

There should be discernable text

### Actual behaviour

There isn't

### Detailed description

Buttons do not have name or text.

Title: WCAG 4.1.2: Ensures buttons have discernible text (.close)
Tags: Accessibility, WCAG 4.1.2, button-name

Issue: Ensures buttons have discernible text (button-name - https://accessibilityinsights.io/info-examples/web/button-name)

Target application: Notifications - Mastodon - https://mastodon.social/notifications

Element path: .close

Snippet: <button type="button" aria-label="" title="" class="close icon-button" tabindex="0" style="font-size: 18px; width: 23.1429px; height: 23.1429px; line-height: 18px;"><i role="img" class="fa fa-close fa-fw" aria-hidden="true"></i> </button>

How to fix: 
Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has an empty title attribute
  Element's default semantics were not overridden with role="none" or role="presentation"

Environment: Chrome version 107.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.35.0 (axe-core 4.4.1), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

### Specifications

https://mastodon.social/notifications

**Comment by Gargron** [comment]
Could you clarify which button it is?

**Comment by mgifford** [comment]
Wait, you want to be able to reproduce this error? 

Sorry. First button here;

`<div class="navigation-bar__actions"><button type="button" aria-label="" title="" class="close icon-button" tabindex="0" style="font-size: 18px; width: 23.1429px; height: 23.1429px; line-height: 18px;"><i role="img" class="fa fa-close fa-fw" aria-hidden="true"></i> </button><div class="compose__action-bar"><div class="compose__action-bar-dropdown"><button type="button" aria-label="Menu" title="Menu" class="icon-button" tabindex="0" style="font-size: 18px; width: 23.1429px; height: 23.1429px; line-height: 18px;"><i role="img" class="fa fa-ellipsis-v fa-fw" aria-hidden="true"></i> </button></div></div></div>`

<img width="1725" alt="Screen Shot 2022-11-11 at 5 29 09 PM" src="https://user-images.githubusercontent.com/116832/201439786-4a0725d3-c606-4e2d-a7e4-3834ba822dc6.png">

**Comment by Gargron** [comment]
Oh, I see. It’s actually not immediately obvious to me what that button does, it seems to not be displayed by default and could be a remnant of an older feature that got removed or stopped working. Thanks for pointing it out!

**Comment by ClearlyClaire** [comment]
This specific button seems long gone, though there is another buttons that doesn't have a label when the search box is empty (in all cases, clicking the button clears the search bar, but the icon looks different and has a label only when the search box is non-empty).

The visibility icons also miss a label.
